### PR TITLE
auth: keep old revision in 'NewAuthStore'

### PR DESCRIPTION
**Pull request by gyuho**
When there's no changes yet (right after auth
store initialization), we should commit old revision.

Fix https://github.com/coreos/etcd/issues/7359.

Seems like `commitRevision` was introduced long time ago,
and we recently fixed it to have correct revision in `authStore`,
so it starts failing in our hash tests.

Do we have any reason to bump up revision for every `commitRevision` call,
even around store initialization?

### CODE REVIEW

File: auth/store.go [hunk 1]
```diff
@@ -916,7 +916,7 @@ func NewAuthStore(be backend.Backend, indexWaiter func(uint64) <-chan struct{})
 		as.simpleTokenKeeper = NewSimpleTokenTTLKeeper(newDeleterFunc(as))
 	}
 
-	as.commitRevision(tx)
+	as.commit(tx)
```
Comment (heyitsanthony) [inline auth/store.go]: I think the initial version needs to commit with rev=1. Maybe:
```go
if as.revision == 0 {
    as.commit()
}
```

File: auth/store_test.go [hunk 2]
```diff
@@ -38,6 +38,33 @@ func dummyIndexWaiter(index uint64) <-chan struct{} {
 	return ch
 }
 
+// TestNewAuthStoreRevision ensures newly auth store
```
Comment (heyitsanthony) [inline auth/store_test.go]: there probably should be an integration test that restarts the etcd server and checks the hash stays the same / fails without this patch

**Comment by gyuho** [comment]
@heyitsanthony PTAL. Thanks.

**Comment by heyitsanthony** [comment]
lgtm defer to @mitake

**Comment by mitake** [comment]
> Do we have any reason to bump up revision for every commitRevision call, even around store initialization?

I added the `commitRevision()` call because the initialization phase can create buckets for the metadata. But it wouldn't be necessarily. Removing it won't produce problems.

**Comment by mitake** [comment]
@gyuho BTW can I ignore the fail of jenkins-proxy-ci? It seems to be the internal problem of the server. Or could you restart the CI?

**Comment by gyuho** [comment]
Yeah proxy CI doesn't seem related to this fix.

Thanks.

**Pull request merged**